### PR TITLE
[WIP] doc: add MAINTAINERS.md

### PR DESCRIPTION
**Pull request by ghost**
Picking up https://github.com/bitcoin/bitcoin/pull/25839 (labeled "Up for grabs")

Addressed all the comments from previous pull request, made a few other changes and added @JeremyRubin as co-author.

### CODE REVIEW

File: MAINTAINERS.md
Comment (MarcoFalke) [inline MAINTAINERS.md:35]: I don't think a single maintainer in the past has been picked based on such a thread, so it seems weird to start prescribing it now and mentioning it seems misleading.

**Review by maflcko — APPROVED** [review]
Please make sure the CI lint passes

File: MAINTAINERS.md
Comment (MarcoFalke) [inline MAINTAINERS.md:19]: A maintainer decision isn't final. It is merely a piece before post-merge review and post-merge testing happens. Also, there is a release process and the every user is free to use their own judgement whether and what software they want to run.

**Comment by michaelfolkson** [comment]
I don't really see the point of this. What is this doc trying to achieve? What real world problems have you identified that this document provides a solution for?

If I'm forced I'll open an alternative PR with some wording on what maintainers do to nip this in the bud but it basically boils down to maintainers merge pull requests, a long term contributor can put their name forward to be a maintainer in a IRC meeting and open a pull request like Gloria, Vasil have done. It isn't going to be a particularly interesting or informative doc.

On the Vasil issue I think we're in agreement(?) @1440000bytes but I don't know how for example this document would resolve the Vasil issue in any way. Two maintainers are refusing to comment (or choosing to ignore the PR). Hence this document is pretty useless in solving the Vasil issue. If someone wants to put their name forward to be a maintainer they can in the IRC meeting. If they are a long term contributor they will know about the IRC meeting already and will be able to assess whether putting their name forward is appropriate or not. They won't need to refer to a document like this.

File: MAINTAINERS.md
Comment (1440000bytes) [reply MAINTAINERS.md]: Removed 'call for maintainers' section and replaced by nomination in IRC channel. However, it was suggested by @ariard as an improvement that I agree with and could be discussed in next IRC meeting. It can be added later if enough contributors agree with it.

File: MAINTAINERS.md
Comment (1440000bytes) [reply MAINTAINERS.md]: I copied this part from https://bitcoincore.org/en/about/. Rephrased 'final check' and added a sentence based on your suggestion.

**Comment by 1440000bytes** [comment]
> I don't really see the point of this. What is this doc trying to achieve? What real world problems have you identified that this document provides a solution for?

This document contains list of present maintainers, their role and selection process. Goal is to document things (related to maintainers for bitcoin core) that everyone agrees upon and improve with time.

This isn't a new practice and you might find lot of open source projects with such file if you search `maintainers.md github` in a search engine. There is a [blog post](https://github.com/readme/guides/maintainer-monolith) about trying to solve related issues and start with `maintainers.md`.

I'll quote one line from it:

"**MAINTAINERS.md** _is the first step towards finding human solutions to human problems_"

> If I'm forced I'll open an alternative PR with some wording on what maintainers do to nip this in the bud but it basically boils down to maintainers merge pull requests, a long term contributor can put their name forward to be a maintainer in a IRC meeting and open a pull request like Gloria, Vasil have done. It isn't going to be a particularly interesting or informative doc.

You are free to open another pull request. 

https://github.com/bitcoin/bitcoin/pull/25839 was closed on Oct 3, 2022 and labelled 'up for grabs' since then. I picked it up based on a suggestion by @MarcoFalke in a [comment](https://github.com/bitcoin/bitcoin/issues/26751#issuecomment-1373656594).

> On the Vasil issue I think we're in agreement(?) @1440000bytes but I don't know how for example this document would resolve the Vasil issue in any way. Two maintainers are refusing to comment (or choosing to ignore the PR). Hence this document is pretty useless in solving the Vasil issue. If someone wants to put their name forward to be a maintainer they can in the IRC meeting. If they are a long term contributor they will know about the IRC meeting already and will be able to assess whether putting their name forward is appropriate or not. They won't need to refer to a document like this.

I have added both pull requests as [topic for next weekly bitcoin core IRC meeting](https://bitcoin-irc.chaincode.com/bitcoin-core-dev/2023-01-11#883146;). This document is public information which helps every new contributor to understand the process. I don't see any downsides of being transparent and document basic things.

**Comment by michaelfolkson** [comment]
> This document contains list of present maintainers, their role and selection process. Goal is to document things (related to maintainers for bitcoin core) that everyone agrees upon and improve with time.

A list of present maintainers and their scope in a doc can't hurt. You keep using strange wording though. It isn't a "selection process" is it? Someone puts their name forward, it is discussed at a IRC meeting, they open a pull request. It seems like you keep wanting to shoehorn this open source project into a formal company structure with a board of directors, an election process and all this other stuff that is totally unnecessary.

> You are free to open another pull request.

Ok I consider myself forced.

**Comment by 1440000bytes** [comment]
> Please make sure the CI lint passes

Fixed

**Comment by michaelfolkson** [comment]
> Ok I consider myself forced.

I won't be opening an alternative PR. This PR can be reviewed as is.

File: MAINTAINERS.md
Comment (vnprc) [inline MAINTAINERS.md:43]: Typo: "maintainer" should be "maintainers"

Comment (1440000bytes) [reply MAINTAINERS.md]: Fixed

**Comment by 1440000bytes** [comment]
> A list of present maintainers and their scope in a doc can't hurt. You keep using strange wording though. It isn't a "selection process" is it?

@michaelfolkson Replaced 'select' with 'add'

File: MAINTAINERS.md
Comment (jonatack) [inline MAINTAINERS.md:5]: I'm unsure if this section is a good idea, though perhaps not for a good reason.  In general, people (e.g. journalists and non-contributors to the project) in my experience overestimate the role of maintainers (i.e. as power/control/gatekeepers rather than implementing the rough consensus of reviewers), and conversely, they undervalue the key vital role of the contributors who are doing consistent review and testing that shape and determine rough consensus.

Also, these roles may be fluid, require updating here, and "General" before each seems redundant.

File: MAINTAINERS.md
Comment (jonatack) [inline MAINTAINERS.md:22]:
```suggestion
The maintainers’ role is by rough consensus of project contributors.
```

File: MAINTAINERS.md
Comment (jonatack) [inline MAINTAINERS.md:19]: perhaps: "They perform a janitorial role of implementing, by merging patches, the rough consensus of reviewers. Maintainers also make sure..."

File: MAINTAINERS.md
Comment (jonatack) [inline MAINTAINERS.md:31]:
```suggestion
However, it may be acceptable for maintainers to occasionally merge their own changes if the pull request
```

File: MAINTAINERS.md
Comment (jonatack) [inline MAINTAINERS.md:34]:
```suggestion
## Nomination and Addition of New Maintainers
```

File: MAINTAINERS.md
Comment (jonatack) [inline MAINTAINERS.md:36]: I think the word "contributor" includes maintainers.
```suggestion
Contributors can nominate a long-term contributor
```

File: MAINTAINERS.md
Comment (jonatack) [inline MAINTAINERS.md:38]:
```suggestion
role of a maintainer. Contributors are free to self-nominate as well.
```

File: MAINTAINERS.md
Comment (jonatack) [inline MAINTAINERS.md:40]:
```suggestion
Maintainers are added by a new maintainer opening a pull request to add their
```

File: MAINTAINERS.md
Comment (jonatack) [inline MAINTAINERS.md:46]:
```suggestion
but the actual decision-making process occurs in the pull request.
```

File: MAINTAINERS.md
Comment (jonatack) [inline MAINTAINERS.md:43]:
```suggestion
is reviewed by the community and merged by one of the maintainers based on rough consensus of the reviewing contributors. Before opening
```

**Review by jonatack — COMMENTED** [review]
A few suggestions from which to pick and choose.

File: CONTRIBUTING.md
Comment (1440000bytes) [reply CONTRIBUTING.md:12]: Done

File: MAINTAINERS.md
Comment (1440000bytes) [reply MAINTAINERS.md:58]: Done

File: MAINTAINERS.md
Comment (1440000bytes) [reply MAINTAINERS.md:43]: Done

File: MAINTAINERS.md
Comment (1440000bytes) [reply MAINTAINERS.md:46]: Done

File: MAINTAINERS.md
Comment (1440000bytes) [reply MAINTAINERS.md:40]: Done

File: MAINTAINERS.md
Comment (1440000bytes) [reply MAINTAINERS.md:38]: Done

File: MAINTAINERS.md
Comment (1440000bytes) [reply MAINTAINERS.md:36]: Done

File: MAINTAINERS.md
Comment (1440000bytes) [reply MAINTAINERS.md:34]: Done

File: MAINTAINERS.md
Comment (1440000bytes) [reply MAINTAINERS.md:19]: I used this sentence, let me know if it conveys the correct meaning: 

"They perform a janitorial role of implementing the rough consensus of reviewers by merging patches."

File: MAINTAINERS.md
Comment (1440000bytes) [reply MAINTAINERS.md]: Done

File: MAINTAINERS.md
Comment (1440000bytes) [reply MAINTAINERS.md:5]: - Removed 'Focus Area' column 
- I understand your point about journalists however they would anyway find these names from other links which maybe misleading, outdated etc. Role of maintainers and contributors is defined in next section with 'rough consensus' part mentioned. A `maintainers.md` file without names of maintainers would be incomplete IMO

**Pull request closed without merging**
